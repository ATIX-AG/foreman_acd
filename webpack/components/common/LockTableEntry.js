import React from 'react';
import PropTypes from 'prop-types';
import {
  Icon,
  Button,
} from 'patternfly-react';

const LockTableEntry = ({
  hidden,
  disabled,
  onLockTableEntry,
  additionalData,
}) =>{
  if (hidden) {
    return null;
  }

  let lockButton;
  let lockButtonTitle;

  if (additionalData.rowData.locked === true) {
    lockButton = 'locked';

    if (disabled === true) {
      lockButtonTitle = 'this entry is locked';
    } else {
      lockButtonTitle = 'unlock this entry';
    }
  } else {
    lockButton = 'unlocked';

    if (disabled === true) {
      lockButtonTitle = 'this entry is unlocked';
    } else {
      lockButtonTitle = 'lock this entry';
    }
  }

  return (
    <span>
      <Button
        bsStyle="default"
        disabled={disabled}
        onClick={() => onLockTableEntry(additionalData) }
      >
        <Icon type="pf" name={lockButton} title={lockButtonTitle} />
      </Button>
    </span>
  );
};

LockTableEntry.propTypes = {
  hidden: PropTypes.bool,
  disabled: PropTypes.bool.isRequired,
  onLockTableEntry: PropTypes.func.isRequired,
  additionalData: PropTypes.object.isRequired,
};

export default LockTableEntry;
