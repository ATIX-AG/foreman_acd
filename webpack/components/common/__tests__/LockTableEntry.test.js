import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import LockTableEntry from '../LockTableEntry';

const noop = () => {};

const fixtures = {
  'should render unlock table entry': {
    hidden: false,
    disabled: false,
    onLockTableEntry: noop,
    additionalData: { rowData: { locked: false } },
  },
  'should render locked table entry': {
    hidden: false,
    disabled: false,
    onLockTableEntry: noop,
    additionalData: { rowData: { locked: true } },
  },
  'should render hidden button': {
    hidden: true,
    disabled: false,
    onLockTableEntry: noop,
    additionalData: {},
  },
  'should render disabled button': {
    hidden: false,
    disabled: true,
    onLockTableEntry: noop,
    additionalData: { rowData: { locked: false } },
  },
};

describe('LockTableEntry', () =>
  testComponentSnapshotsWithFixtures(LockTableEntry, fixtures));
