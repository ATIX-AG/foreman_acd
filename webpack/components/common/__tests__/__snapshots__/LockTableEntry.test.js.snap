// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LockTableEntry should render disabled button 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <Icon
      name="unlocked"
      title="this entry is unlocked"
      type="pf"
    />
  </Button>
</span>
`;

exports[`LockTableEntry should render hidden button 1`] = `""`;

exports[`LockTableEntry should render locked table entry 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    <Icon
      name="locked"
      title="unlock this entry"
      type="pf"
    />
  </Button>
</span>
`;

exports[`LockTableEntry should render unlock table entry 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    <Icon
      name="unlocked"
      title="lock this entry"
      type="pf"
    />
  </Button>
</span>
`;
